test/plugins/sensor_demo/custom_system/CustomSystem.hh

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef IGNITION_GAZEBO_SYSTEMS_CUSTOMSENSORS_HH_
+#define IGNITION_GAZEBO_SYSTEMS_CUSTOMSENSORS_HH_
+
+#include <memory>
+#include <string>
+
+#include <ignition/gazebo/config.hh>
+#include <ignition/gazebo/System.hh>
+#include <sdf/Sensor.hh>
+
+namespace ignition
+{
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace systems
+{
+  // Forward declarations.
+  class SensorsPrivate;
+
+  /// \class Sensors
+  /// \brief An example system that manages custom rendering sensors.
+  ///
+  /// ## System Parameters
+  ///
+  /// - `<render_engine>` Name of the render engine, such as 'ogre' or 'ogre2'.
+  /// - `<background_color>` Color used for the scene's background. This
+  /// will override the background color specified in a world's SDF <scene>
+  /// element. This background color is used by sensors, not the GUI.
+  /// - `<ambient_light>` Color used for the scene's ambient light. This
+  /// will override the ambient value specified in a world's SDF <scene>
+  /// element. This ambient light is used by sensors, not the GUI.
+  ///
+  class Sensors:
+    public System,
+    public ISystemConfigure,
+    public ISystemUpdate,
```

Comment:
`ISystemUpdate` is generally reserved for physics updates (see https://ignitionrobotics.org/api/gazebo/4.6/createsystemplugins.html). Can this be changed ti `ISystemPreUpdate`?

Reply:
Done. I used the system 'sensors' plugin as a base for this ``CustomSystem``. Should this line be changed as well in the original plugin then ? https://github.com/ignitionrobotics/ign-gazebo/blob/9cd15fc941b0e9435abb6fa3598b54527009a074/src/systems/sensors/Sensors.hh#L56